src/tk/wurst_client/mods/PlayerFinderMod.java

```diff
@@ -85,30 +104,77 @@ public void onReceivedPacket(PacketInputEvent event)
 		if(mc.thePlayer == null)
 			return;
 		Packet packet = event.getPacket();
-		if(packet instanceof SPacketEffect)
+		if(packet instanceof SPacketBlockAction)
+		{
+			BlockPos pos = ((SPacketBlockAction)packet).getBlockPosition();
+			this.listenToBlockPosition(pos);
+		}else if(packet instanceof SPacketBlockChange)
+		{
+			BlockPos pos = ((SPacketBlockChange)packet).getBlockPosition();
+			this.listenToBlockPosition(pos);
+		}else if(packet instanceof SPacketBlockBreakAnim)
+		{
+			BlockPos pos = ((SPacketBlockBreakAnim)packet).getPosition();
+			this.listenToBlockPosition(pos);
+		}else if(packet instanceof SPacketEffect)
 		{
 			SPacketEffect effect = (SPacketEffect)packet;
 			BlockPos pos = effect.getSoundPos();
-			if(BlockUtils.getPlayerBlockDistance(pos) >= 160)
-				blockPos = pos;
+			this.listenToBlockPosition(pos);
 		}else if(packet instanceof SPacketSoundEffect)
 		{
 			SPacketSoundEffect sound = (SPacketSoundEffect)packet;
 			BlockPos pos =
-				new BlockPos(sound.getX(), sound.getY(),
-					sound.getZ());
-			if(BlockUtils.getPlayerBlockDistance(pos) >= 160)
-				blockPos = pos;
+				new BlockPos(sound.getX(), sound.getY(), sound.getZ());
+			this.listenToBlockPosition(pos);
 		}else if(packet instanceof SPacketSpawnGlobalEntity)
 		{
 			SPacketSpawnGlobalEntity lightning =
 				(SPacketSpawnGlobalEntity)packet;
-			BlockPos pos =
-				new BlockPos(lightning.getX() / 32D,
-					lightning.getY() / 32D,
-					lightning.getZ() / 32D);
-			if(BlockUtils.getPlayerBlockDistance(pos) >= 160)
-				blockPos = pos;
+			BlockPos pos = new BlockPos(lightning.getX() / 32D,
+				lightning.getY() / 32D, lightning.getZ() / 32D);
+			this.listenToBlockPosition(pos);
+		}
+	}
+	
+	// Used to add a position and time added to the hash map if the pos is far
+	// enough
+	public void listenToBlockPosition(BlockPos pos)
+	{
+		if(this.isAGoodBlockPos(pos))
+		{
+			this.blocksPos.put(pos, System.currentTimeMillis());
+		}
+	}
+	
+	// Checks if the given pos isn't too close to another listened pos
+	// and it is not too close to the player
+	// prevents a lot of false positives and "tracers spam"
+	private boolean isAGoodBlockPos(BlockPos testPos)
```

Comment:
Hi! Thanks for taking time to review my code, I will make some changes to improve it.
I have some question too, I'm ask them in the other reviews.